### PR TITLE
Fix AssetHttpPrefix When Using useDeprecatedIncorrectCSSProcessing

### PR DESCRIPTION
**Pull request by mahirshah**
- previously when a lazy engine used useDeprecatedIncorrectCSSProcessing, the assetHttpPrefix would resolve to the app level assets folder
- this commit resolves the assetHttpPrefix to the engine-dist folder regardless of the 
useDeprecatedIncorrectCssProcessing flag because useDeprecatedIncorrectCssProcessing should only influence app specific styles

### CODE REVIEW

File: tests/dummy/lib/lazy/index.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+  eyeglass: {
+    discover: true,
+    assets: assetsDir,
+  },
 
   lazyLoading: {
     enabled: true
   },
 
+  useDeprecatedIncorrectCSSProcessing: true,
```
Comment (dnachev) [inline tests/dummy/lib/lazy/index.js]: The code no longer consider this flag and the test will pass with or without it, therefore we should not probably include it.

The situation would be different if we have a test which does depend on the flag - assets in `app/styles` assets should be prefixed with `/assets` for a lazy engine. The concern in this case is this is a flag which is quite old now and no one should be using it, therefore the value of the test is limited.

However, a test, which validates that the assets from `app/styles` end up under `/engines-dist/lazy/...` will validate the behavior, which we must maintain in future, therefore it is a valuable test to add. I don't see this additional test as a requirement, so we should be fine with submitting the change without it.

Comment (mahirshah) [reply tests/dummy/lib/lazy/index.js]: Makes sense -- removed this flag

**Comment by stefanpenner** [comment]
thanks for working on this, two things:

* we gotta get CI green
* lets confirm output of a real app ends up being correct

**Comment by stefanpenner** [comment]
released as v5.0.0 🎉

**Pull request merged**